packages/@lwc/synthetic-shadow/src/shared/static-html-collection.ts

```diff
@@ -47,76 +42,27 @@ StaticHTMLCollection.prototype = create(HTMLCollection.prototype, {
         enumerable: true,
         configurable: true,
         value(name: string) {
-            if (isValidHTMLCollectionName(name) && this[name]) {
-                return this[name];
+            if (name === '') {
+                return null;
             }
-            const items = getHiddenField(this, Items);
+
+            const items = getHiddenField(this, Items) as Element[];
             // Note: loop in reverse so that the first named item matches the named property
             for (let len = items.length - 1; len >= 0; len -= 1) {
                 const item = items[len];
-                const nodeName = getNodeHTMLCollectionName(item);
-                if (nodeName === name) {
+
+                if (
+                    name === getAttribute.call(item, 'id') ||
```

Comment:
this is not the same as before, but I think this is actually fixing a bug in the previous code, in the pass, if you had an id and name in the same element, only the id will match, now it also checks the name. Good!

Reply:
Yes, I also spotted this issue.